include/picongpu/particles/flylite/IFlyLite.hpp

```diff
@@ -67,9 +67,16 @@ namespace flylite
         )
         {
             boost::ignore_unused( ionSpeciesName, currentStep );
-
+            /* The compiler is allowed to evaluate an expression those not depends on a template parameter
+             * even if the class is never instantiated. In that case static assert is always
+             * evaluated (e.g. with clang), this results in an error if the condition is false.
+             * http://www.boost.org/doc/libs/1_60_0/doc/html/boost_staticassert.html
+             *
+             * A workaround is to add a template dependency to the expression.
+             * `sizeof(ANY_TYPE) != 0` is always true and defers the evaluation.
+             */
             PMACC_STATIC_ASSERT_MSG(
-                false,
+                false && sizeof(T_IonSpecies) != 0,
```

Comment:
You could also use `alpaka::meta::DependentFalseType<T_IonSpecies>`

Reply:
good to know, I will use in in a separate PR because we have used a few of this code constructs in PMacc and PIConGPU.